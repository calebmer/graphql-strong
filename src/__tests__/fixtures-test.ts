import * as path from 'path'
import * as fs from 'fs'
import * as ts from 'typescript'

const compilerOptions: ts.CompilerOptions = {
  target: ts.ScriptTarget.ES2015,
  noEmit: true,
  noImplicitAny: true,
  strictNullChecks: true,
  include: [path.resolve(__dirname, '../../src')]
}

const fixturesDir = path.resolve(__dirname, 'fixtures')

fs.readdirSync(fixturesDir).forEach(name => {
  test(name, () => {
    const fixtureFileName = path.join(fixturesDir, name)
    const program = ts.createProgram([fixtureFileName], compilerOptions)
    const emitResult = program.emit()
    const allDiagnostics = [...ts.getPreEmitDiagnostics(program), ...emitResult.diagnostics]
    expect(allDiagnostics.map(({ file, ...rest }) => rest)).toMatchSnapshot()
  })
})
