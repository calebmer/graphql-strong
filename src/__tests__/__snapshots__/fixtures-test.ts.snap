exports[`test object-fixture.ts 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2345,
    "length": 74,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<number>; resolve: ({url}: Image) => string; }\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, string>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'type\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'StrongGraphQLInputOutputType<number>\' is not assignable to type \'StrongGraphQLOutputType<string> | (() => StrongGraphQLOutputType<string>)\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'StrongGraphQLInputOutputType<number>\' is not assignable to type \'() => StrongGraphQLOutputType<string>\'.",
            "next": Object {
              "category": 1,
              "code": 2658,
              "messageText": "Type \'StrongGraphQLInputOutputType<number>\' provides no match for the signature \'(): StrongGraphQLOutputType<string>\'",
              "next": undefined,
            },
          },
        },
      },
    },
    "start": 1539,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 75,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Person, {}>; resolve: ({url}: Image) => string; }\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, string>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'type\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'StrongGraphQLObjectType<Person, {}>\' is not assignable to type \'StrongGraphQLOutputType<string> | (() => StrongGraphQLOutputType<string>)\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'StrongGraphQLObjectType<Person, {}>\' is not assignable to type \'() => StrongGraphQLOutputType<string>\'.",
            "next": Object {
              "category": 1,
              "code": 2658,
              "messageText": "Type \'StrongGraphQLObjectType<Person, {}>\' provides no match for the signature \'(): StrongGraphQLOutputType<string>\'",
              "next": undefined,
            },
          },
        },
      },
    },
    "start": 1666,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 112,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputType<string>; resolve: ({url}: Image) => string; }\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, string>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'type\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'StrongGraphQLInputType<string>\' is not assignable to type \'StrongGraphQLOutputType<string> | (() => StrongGraphQLOutputType<string>)\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'StrongGraphQLInputType<string>\' is not assignable to type \'() => StrongGraphQLOutputType<string>\'.",
            "next": Object {
              "category": 1,
              "code": 2658,
              "messageText": "Type \'StrongGraphQLInputType<string>\' provides no match for the signature \'(): StrongGraphQLOutputType<string>\'",
              "next": undefined,
            },
          },
        },
      },
    },
    "start": 1820,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 88,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<number>; resolve: ({size}: Image) => string; }\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, number | null | undefined>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'resolve\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'({size}: Image) => string\' is not assignable to type \'(source: Image, args: {}, context: {}) => number | Promise<number | null | undefined> | null | un...\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'string\' is not assignable to type \'number | Promise<number | null | undefined> | null | undefined\'.",
            "next": undefined,
          },
        },
      },
    },
    "start": 2190,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 77,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<number>; resolve: ({size}: Image) => number | ...\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, number>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'resolve\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'({size}: Image) => number | undefined\' is not assignable to type \'(source: Image, args: {}, context: {}) => number | Promise<number>\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'number | undefined\' is not assignable to type \'number | Promise<number>\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'undefined\' is not assignable to type \'number | Promise<number>\'.",
              "next": undefined,
            },
          },
        },
      },
    },
    "start": 2339,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 95,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<string | null | undefined>; resolve: ({url}: I...\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, string>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'type\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'StrongGraphQLOutputType<string> | (() => StrongGraphQLOutputType<string>)\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'() => StrongGraphQLOutputType<string>\'.",
            "next": Object {
              "category": 1,
              "code": 2658,
              "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' provides no match for the signature \'(): StrongGraphQLOutputType<string>\'",
              "next": undefined,
            },
          },
        },
      },
    },
    "start": 2693,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 313,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<string>; args: { width: { type: StrongGraphQLI...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLInputOutputType<string>; args: { width: { type: StrongGraphQLI...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'type\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'StrongGraphQLInputOutputType<string>\' is not assignable to type \'StrongGraphQLOutputType<Image | null | undefined> | (() => StrongGraphQLOutputType<Image | null |...\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'StrongGraphQLInputOutputType<string>\' is not assignable to type \'() => StrongGraphQLOutputType<Image | null | undefined>\'.",
              "next": Object {
                "category": 1,
                "code": 2658,
                "messageText": "Type \'StrongGraphQLInputOutputType<string>\' provides no match for the signature \'(): StrongGraphQLOutputType<Image | null | undefined>\'",
                "next": undefined,
              },
            },
          },
        },
      },
    },
    "start": 3072,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 311,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<string>; }; height: { type: StrongGraphQLInputOutpu...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<string>; }; height: { type: StrongGraphQLInputOutpu...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2326,
                "messageText": "Types of property \'width\' are incompatible.",
                "next": Object {
                  "category": 1,
                  "code": 2322,
                  "messageText": "Type \'{ type: StrongGraphQLInputOutputType<string>; }\' is not assignable to type \'StrongGraphQLArgConfig<number>\'.",
                  "next": Object {
                    "category": 1,
                    "code": 2326,
                    "messageText": "Types of property \'type\' are incompatible.",
                    "next": Object {
                      "category": 1,
                      "code": 2322,
                      "messageText": "Type \'StrongGraphQLInputOutputType<string>\' is not assignable to type \'StrongGraphQLInputType<number>\'.",
                      "next": Object {
                        "category": 1,
                        "code": 2326,
                        "messageText": "Types of property \'nullable\' are incompatible.",
                        "next": Object {
                          "category": 1,
                          "code": 2322,
                          "messageText": "Type \'() => StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'() => StrongGraphQLInputType<number | null | undefined>\'.",
                          "next": Object {
                            "category": 1,
                            "code": 2322,
                            "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'StrongGraphQLInputType<number | null | undefined>\'.",
                            "next": Object {
                              "category": 1,
                              "code": 2326,
                              "messageText": "Types of property \'nullable\' are incompatible.",
                              "next": Object {
                                "category": 1,
                                "code": 2322,
                                "messageText": "Type \'() => StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'() => StrongGraphQLInputType<number | null | undefined>\'.",
                                "next": Object {
                                  "category": 1,
                                  "code": 2322,
                                  "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'StrongGraphQLInputType<number | null | undefined>\'.",
                                  "next": undefined,
                                },
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
    "start": 3488,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 311,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number>; }; height: { type: StrongGraphQLInputOutpu...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number>; }; height: { type: StrongGraphQLInputOutpu...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2326,
                "messageText": "Types of property \'height\' are incompatible.",
                "next": Object {
                  "category": 1,
                  "code": 2322,
                  "messageText": "Type \'{ type: StrongGraphQLInputOutputType<string | null | undefined>; }\' is not assignable to type \'StrongGraphQLArgConfig<number | undefined>\'.",
                  "next": Object {
                    "category": 1,
                    "code": 2326,
                    "messageText": "Types of property \'type\' are incompatible.",
                    "next": Object {
                      "category": 1,
                      "code": 2322,
                      "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'StrongGraphQLInputType<number | undefined>\'.",
                      "next": Object {
                        "category": 1,
                        "code": 2326,
                        "messageText": "Types of property \'nullable\' are incompatible.",
                        "next": Object {
                          "category": 1,
                          "code": 2322,
                          "messageText": "Type \'() => StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'() => StrongGraphQLInputType<number | null | undefined>\'.",
                          "next": Object {
                            "category": 1,
                            "code": 2322,
                            "messageText": "Type \'StrongGraphQLInputOutputType<string | null | undefined>\' is not assignable to type \'StrongGraphQLInputType<number | null | undefined>\'.",
                            "next": undefined,
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
    "start": 3902,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 170,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; resolve: ({imageURL}: Person, {width, h...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; resolve: ({imageURL}: Person, {width, h...\' is not assignable to type \'{ readonly args: StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>; }\'.",
        "next": Object {
          "category": 1,
          "code": 2324,
          "messageText": "Property \'args\' is missing in type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; resolve: ({imageURL}: Person, {width, h...\'.",
          "next": undefined,
        },
      },
    },
    "start": 4322,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 261,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { height: { type: StrongGraphQLIn...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { height: { type: StrongGraphQLIn...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ height: { type: StrongGraphQLInputOutputType<number | null | undefined>; }; }\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ height: { type: StrongGraphQLInputOutputType<number | null | undefined>; }; }\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2324,
                "messageText": "Property \'width\' is missing in type \'{ height: { type: StrongGraphQLInputOutputType<number | null | undefined>; }; }\'.",
                "next": undefined,
              },
            },
          },
        },
      },
    },
    "start": 4598,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 240,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number>; }; }\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number>; }; }\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2324,
                "messageText": "Property \'height\' is missing in type \'{ width: { type: StrongGraphQLInputOutputType<number>; }; }\'.",
                "next": undefined,
              },
            },
          },
        },
      },
    },
    "start": 4965,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 332,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLInp...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number | null | undefined>; }; height: { type: Stro...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ width: { type: StrongGraphQLInputOutputType<number | null | undefined>; }; height: { type: Stro...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2326,
                "messageText": "Types of property \'width\' are incompatible.",
                "next": Object {
                  "category": 1,
                  "code": 2322,
                  "messageText": "Type \'{ type: StrongGraphQLInputOutputType<number | null | undefined>; }\' is not assignable to type \'StrongGraphQLArgConfig<number>\'.",
                  "next": Object {
                    "category": 1,
                    "code": 2326,
                    "messageText": "Types of property \'type\' are incompatible.",
                    "next": Object {
                      "category": 1,
                      "code": 2322,
                      "messageText": "Type \'StrongGraphQLInputOutputType<number | null | undefined>\' is not assignable to type \'StrongGraphQLInputType<number>\'.",
                      "next": Object {
                        "category": 1,
                        "code": 2326,
                        "messageText": "Types of property \'_strongValue\' are incompatible.",
                        "next": Object {
                          "category": 1,
                          "code": 2322,
                          "messageText": "Type \'number | null | undefined\' is not assignable to type \'number | null\'.",
                          "next": Object {
                            "category": 1,
                            "code": 2322,
                            "messageText": "Type \'undefined\' is not assignable to type \'number | null\'.",
                            "next": undefined,
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
    "start": 5324,
  },
  Object {
    "category": 1,
    "code": 2459,
    "length": 5,
    "messageText": "Type \'{}\' has no property \'width\' and no string index signature.",
    "start": 5929,
  },
  Object {
    "category": 1,
    "code": 2459,
    "length": 6,
    "messageText": "Type \'{}\' has no property \'height\' and no string index signature.",
    "start": 5936,
  },
  Object {
    "category": 1,
    "code": 2459,
    "length": 1,
    "messageText": "Type \'{ width: number; height: number | undefined; }\' has no property \'x\' and no string index signature.",
    "start": 6985,
  },
  Object {
    "category": 1,
    "code": 2459,
    "length": 1,
    "messageText": "Type \'{ width: number; height: number | undefined; }\' has no property \'y\' and no string index signature.",
    "start": 6988,
  },
  Object {
    "category": 1,
    "code": 2459,
    "length": 1,
    "messageText": "Type \'{ width: number; height: number | undefined; }\' has no property \'z\' and no string index signature.",
    "start": 6991,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 349,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLOut...\' is not assignable to parameter of type \'StrongGraphQLFieldConfigWithArgs<Person, { width: number; height: number | undefined; }, {}, Imag...\'.",
      "next": Object {
        "category": 1,
        "code": 2322,
        "messageText": "Type \'{ name: string; type: StrongGraphQLObjectType<Image, {}>; args: { width: { type: StrongGraphQLOut...\' is not assignable to type \'StrongGraphQLFieldConfig<Person, { width: number; height: number | undefined; }, {}, Image | null...\'.",
        "next": Object {
          "category": 1,
          "code": 2326,
          "messageText": "Types of property \'args\' are incompatible.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'{ width: { type: StrongGraphQLOutputType<number>; }; height: { type: StrongGraphQLInputOutputType...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }> | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'{ width: { type: StrongGraphQLOutputType<number>; }; height: { type: StrongGraphQLInputOutputType...\' is not assignable to type \'StrongGraphQLArgsConfig<{ width: number; height: number | undefined; }>\'.",
              "next": Object {
                "category": 1,
                "code": 2326,
                "messageText": "Types of property \'width\' are incompatible.",
                "next": Object {
                  "category": 1,
                  "code": 2322,
                  "messageText": "Type \'{ type: StrongGraphQLOutputType<number>; }\' is not assignable to type \'StrongGraphQLArgConfig<number>\'.",
                  "next": Object {
                    "category": 1,
                    "code": 2326,
                    "messageText": "Types of property \'type\' are incompatible.",
                    "next": Object {
                      "category": 1,
                      "code": 2322,
                      "messageText": "Type \'StrongGraphQLOutputType<number>\' is not assignable to type \'StrongGraphQLInputType<number>\'.",
                      "next": Object {
                        "category": 1,
                        "code": 2324,
                        "messageText": "Property \'_strongInputType\' is missing in type \'StrongGraphQLOutputType<number>\'.",
                        "next": undefined,
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
    "start": 7639,
  },
  Object {
    "category": 1,
    "code": 2339,
    "length": 12,
    "messageText": "Property \'ImageService\' does not exist on type \'{}\'.",
    "start": 8868,
  },
  Object {
    "category": 1,
    "code": 2448,
    "length": 4,
    "messageText": "Block-scoped variable \'type\' used before its declaration.",
    "start": 9113,
  },
  Object {
    "category": 1,
    "code": 2454,
    "length": 4,
    "messageText": "Variable \'type\' is used before being assigned.",
    "start": 9113,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 97,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; type: StrongGraphQLInputOutputType<number>; resolve: () => Promise<string>; }\' is not assignable to parameter of type \'StrongGraphQLFieldConfig<Image, {}, {}, number | null | undefined>\'.",
      "next": Object {
        "category": 1,
        "code": 2326,
        "messageText": "Types of property \'resolve\' are incompatible.",
        "next": Object {
          "category": 1,
          "code": 2322,
          "messageText": "Type \'() => Promise<string>\' is not assignable to type \'(source: Image, args: {}, context: {}) => number | Promise<number | null | undefined> | null | un...\'.",
          "next": Object {
            "category": 1,
            "code": 2322,
            "messageText": "Type \'Promise<string>\' is not assignable to type \'number | Promise<number | null | undefined> | null | undefined\'.",
            "next": Object {
              "category": 1,
              "code": 2322,
              "messageText": "Type \'Promise<string>\' is not assignable to type \'Promise<number | null | undefined>\'.",
              "next": Object {
                "category": 1,
                "code": 2322,
                "messageText": "Type \'string\' is not assignable to type \'number | null | undefined\'.",
                "next": undefined,
              },
            },
          },
        },
      },
    },
    "start": 9631,
  },
]
`;

exports[`test scalar-fixture.ts 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2345,
    "length": 103,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; serialize: (value: string) => number; parseValue: (value: number) => string; }\' is not assignable to parameter of type \'StrongGraphQLScalarTypeConfigWithInput<string, number>\'.",
      "next": Object {
        "category": 1,
        "code": 2324,
        "messageText": "Property \'parseLiteral\' is missing in type \'{ name: string; serialize: (value: string) => number; parseValue: (value: number) => string; }\'.",
        "next": undefined,
      },
    },
    "start": 347,
  },
  Object {
    "category": 1,
    "code": 2345,
    "length": 102,
    "messageText": Object {
      "category": 1,
      "code": 2345,
      "messageText": "Argument of type \'{ name: string; serialize: (value: string) => number; parseLiteral: (value: ValueNode) => \"Object...\' is not assignable to parameter of type \'StrongGraphQLScalarTypeConfigWithInput<string, number>\'.",
      "next": Object {
        "category": 1,
        "code": 2324,
        "messageText": "Property \'parseValue\' is missing in type \'{ name: string; serialize: (value: string) => number; parseLiteral: (value: ValueNode) => \"Object...\'.",
        "next": undefined,
      },
    },
    "start": 556,
  },
  Object {
    "category": 1,
    "code": 2322,
    "length": 5,
    "messageText": Object {
      "category": 1,
      "code": 2322,
      "messageText": "Type \'StrongGraphQLOutputType<string>\' is not assignable to type \'StrongGraphQLInputType<string>\'.",
      "next": Object {
        "category": 1,
        "code": 2324,
        "messageText": "Property \'_strongInputType\' is missing in type \'StrongGraphQLOutputType<string>\'.",
        "next": undefined,
      },
    },
    "start": 869,
  },
]
`;

exports[`test wrap-fixture.ts 1`] = `
Array [
  Object {
    "category": 1,
    "code": 2322,
    "length": 6,
    "messageText": Object {
      "category": 1,
      "code": 2322,
      "messageText": "Type \'StrongGraphQLInputType<{}>\' is not assignable to type \'StrongGraphQLOutputType<{}>\'.",
      "next": Object {
        "category": 1,
        "code": 2324,
        "messageText": "Property \'_strongOutputType\' is missing in type \'StrongGraphQLInputType<{}>\'.",
        "next": undefined,
      },
    },
    "start": 399,
  },
  Object {
    "category": 1,
    "code": 2322,
    "length": 6,
    "messageText": "Type \'StrongGraphQLInputType<{}>\' is not assignable to type \'StrongGraphQLOutputType<{}>\'.",
    "start": 612,
  },
  Object {
    "category": 1,
    "code": 2322,
    "length": 5,
    "messageText": Object {
      "category": 1,
      "code": 2322,
      "messageText": "Type \'StrongGraphQLOutputType<{}>\' is not assignable to type \'StrongGraphQLInputType<{}>\'.",
      "next": Object {
        "category": 1,
        "code": 2324,
        "messageText": "Property \'_strongInputType\' is missing in type \'StrongGraphQLOutputType<{}>\'.",
        "next": undefined,
      },
    },
    "start": 771,
  },
  Object {
    "category": 1,
    "code": 2322,
    "length": 5,
    "messageText": "Type \'StrongGraphQLOutputType<{}>\' is not assignable to type \'StrongGraphQLInputType<{}>\'.",
    "start": 979,
  },
]
`;
